src/utils/spread.js

```diff
@@ -47,8 +47,13 @@ export default function spread(
 	const previousElement = elements[firstSpreadIndex - 1];
 
 	if (!previousElement) {
-		code.remove(start, element.start);
-		code.overwrite(element.end, elements[1].start, '.concat( ');
+		// We parenthesize it to handle ternaries like [...a ? b : c].
+		// (We could be smarter about it, but with the parenthesizing having
+		// spread to CallExpression.js, it sounds too much like hard work.)
+		if (start !== element.start) {
+			code.overwrite(start, element.start, '( ');
+		} // else CallExpression called us, and it puts the ( in itself.
```

Comment:
Slightly concerned that the `start !== element.start` might not be robust/explicit enough to match the code paths in `CallExpression` that add the opening parens. Would a parent check be better here?

Reply:
Seems a reasonable concern. I’ve added another commit with what I imagine you would like there. (I’d amend the first commit, but then GitHub doesn’t show a proper diff.)